Mathlib/Computability/NFA.lean

```diff
@@ -66,11 +90,31 @@ theorem evalFrom_nil (S : Set σ) : M.evalFrom S [] = S :=
 theorem evalFrom_singleton (S : Set σ) (a : α) : M.evalFrom S [a] = M.stepSet S a :=
   rfl
 
+@[simp]
+theorem evalFrom_cons (S : Set σ) (a : α) (x : List α) :
+    M.evalFrom S (a :: x) = M.evalFrom (M.stepSet S a) x :=
+  rfl
+
 @[simp]
 theorem evalFrom_append_singleton (S : Set σ) (x : List α) (a : α) :
     M.evalFrom S (x ++ [a]) = M.stepSet (M.evalFrom S x) a := by
   simp only [evalFrom, List.foldl_append, List.foldl_cons, List.foldl_nil]
 
+theorem evalFrom_biUnion {ι : Type*} (t : Set ι) (f : ι → Set σ) (x : List α) :
+    M.evalFrom (⋃ i ∈ t, f i) x = ⋃ i ∈ t, M.evalFrom (f i) x := by
```

Comment:
Do you think this should be simp?

Reply:
Perhaps it should be. I was a bit disinclined towards it by the very next lemma doing `simp [← evalFrom_biUnion]`, but that isn't really enough reason, is it.

Applied `@[simp]` in the next revision.